packages/office-ui-fabric-react/src/components/DetailsList/ShimmeredDetailsList.base.tsx

```diff
@@ -21,141 +21,145 @@ const SHIMMER_INITIAL_ITEMS = 10;
 const DEFAULT_SHIMMER_HEIGHT = 7;
 const SHIMMER_LINE_VS_CELL_WIDTH_RATIO = 0.95;
 
-export class ShimmeredDetailsListBase extends React.Component<IShimmeredDetailsListProps, {}> {
-  private _shimmerItems: null[];
-  private _classNames: IProcessedStyleSet<IShimmeredDetailsListStyles>;
-
-  constructor(props: IShimmeredDetailsListProps) {
-    super(props);
-
-    this._shimmerItems = props.shimmerLines ? new Array(props.shimmerLines) : new Array(SHIMMER_INITIAL_ITEMS);
-  }
-
-  public render(): JSX.Element {
-    const {
-      detailsListStyles,
-      enableShimmer,
-      items,
-      listProps,
-      onRenderCustomPlaceholder,
-      removeFadingOverlay,
-      shimmerLines,
-      styles,
-      theme,
-      ariaLabelForGrid,
-      ariaLabelForShimmer,
-      ...restProps
-    } = this.props;
-
-    const listClassName = listProps && listProps.className;
-
-    this._classNames = getClassNames(styles, {
-      theme: theme!,
-    });
-
-    const newListProps = {
+export function ShimmeredDetailsListBase(props: IShimmeredDetailsListProps): JSX.Element | null {
```

Comment:
Typically we've been doing `export const ShimmeredDetailsListBase: React.FunctionComponent<IShimmeredDetailsListProps> = props => {`